src/metabase/query_processor/middleware/check_features.clj

```diff
@@ -1,29 +1,32 @@
 (ns metabase.query-processor.middleware.check-features
   (:require [metabase.driver :as driver]
+            [metabase.mbql.schema :as mbql.s]
             [metabase.mbql.util :as mbql.u]
             [metabase.query-processor.error-type :as error-type]
+            [metabase.query-processor.store :as qp.store]
             [metabase.util :as u]
             [metabase.util.i18n :refer [tru]]))
 
-;; `assert-driver-supports` doesn't run check when `*driver*` is unbound (e.g., when used in the REPL)
-;; Allows flexibility when composing queries for tests or interactive development
 (defn assert-driver-supports
-  "When `*driver*` is bound, assert that is supports keyword FEATURE."
+  "Assert that the driver/database supports keyword `feature`."
   [feature]
-  (when driver/*driver*
-    (when-not (driver/supports? driver/*driver* feature)
-      (throw (ex-info (tru "{0} is not supported by this driver." (name feature))
-               {:type error-type/unsupported-feature})))))
+  (when-not (driver/database-supports? driver/*driver* feature (qp.store/database))
+    (throw (ex-info (tru "{0} is not supported by this driver." (name feature))
+                    {:type    error-type/unsupported-feature
+                     :feature feature}))))
 
 ;; TODO - definitely a little incomplete. It would be cool if we cool look at the metadata in the schema namespace and
 ;; auto-generate this logic
 (defn- query->required-features [query]
-  (mbql.u/match (:query query)
-    :stddev
-    :standard-deviation-aggregations
+  (into
+   #{}
+   (mbql.u/match (:query query)
+     :stddev
+     :standard-deviation-aggregations
 
-    [:field _ (_ :guard (some-fn :source-field :join-alias))]
-    :foreign-keys))
+     (join :guard (every-pred map? (comp mbql.s/join-strategies :strategy)))
+     (let [{:keys [strategy]} join]
```

Comment:
Am I reading this wrong, or doesn't `assert-driver-supports` throw now, so you wouldn't be filling up that above hash set, just throwing?

Reply:
Yeah you're completely right. I'll fix it